txpipe/lssweights.py

```diff
@@ -0,0 +1,1103 @@
+			self.Ntomo = f['lens'].attrs["nbin_lens"] 
+
+		#output directory for the plots and summary stats
+		output_dir = self.open_output("lss_weight_summary", wrapper=True)
+
+		#load the SP maps, apply the mask, normalize the maps (as needed by the method)
+		self.sys_maps, self.sys_names, self.sys_meta = self.prepare_sys_maps()
+
+		Fmap_list = []
+		for ibin in range(self.Ntomo):
+
+			#compute density vs SP map data vector
+			density_corrs = self.calc_1d_density(ibin)
+
+			#compute covariance of data vector
+			self.calc_covariance(density_corrs) #matrix added to density_corrs
+
+			#compute the weights
+			Fmap, fit_output = self.compute_weights(density_corrs)
+			Fmap_list.append(Fmap)
+
+			weighted_density_corrs = self.calc_1d_density(ibin, Fmap=Fmap)
+			if weighted_density_corrs is not None:
+				weighted_density_corrs.postprocess(density_corrs) #adds needed info from unweighted density_corrs to weighted
+
+			#make summary stats and plots
+			self.summarize(output_dir, density_corrs, weighted_density_corrs, fit_output )
+
+		#save object weights and weight maps
+		self.save_weights(output_dir, Fmap_list)
+
+
+	def read_healsparse(self, map_path, nside):
+		"""
+		returns a healsparse object degraded to nside
+		"""
+		import healsparse
+		import healpy
+
+		# Convert to correct res healpix map
+		m = healsparse.HealSparseMap.read(map_path)
+		return m.degrade(nside)
+
+	def prepare_sys_maps(self):
+		"""
+		By default prepare sysmaps will just load and mask the maps
+
+		subclasses for differnet methods can modify this behaviour 
+		(e.g. adding a normalization of the maps)
+		"""
+		sys_maps, sys_names = self.load_and_mask_sysmaps()
+		sys_meta = {'masked':True,'normed':False}
+		return sys_maps, sys_names, sys_meta
+
+	def get_deltag(self, tomobin):
+		"""
+		convert the ra, dec of the lens sample to pixel number counts
+		"""
+		import healpy as hp
+		import numpy as np  
+		import healsparse as hsp
+
+		with self.open_input("mask", wrapper=True) as map_file:
+			nside = map_file.read_map_info("mask")["nside"]
+			nest = map_file.read_map_info("mask")["nest"]
+			mask = map_file.read_map("mask")
+
+		#load the ra and dec of this lens bins
+		with self.open_input("binned_lens_catalog_unweighted", wrapper=False) as f:
+			ra = f[f"lens/bin_{tomobin}/ra"][:]
+			dec = f[f"lens/bin_{tomobin}/dec"][:]
+			weight = f[f"lens/bin_{tomobin}/weight"][:]
+
+		#pixel ID for each lens galaxy
+		obj_pix = hp.ang2pix(nside,ra,dec,lonlat=True, nest=True)
+
+		Ncounts_bincount = np.bincount(obj_pix, weights=weight)
+		pixels_bincount = np.arange(len(Ncounts_bincount))
+		
+		pixel = pixels_bincount[Ncounts_bincount != 0.]
+		Ncounts = Ncounts_bincount[Ncounts_bincount != 0.]
+
+		if nest:
+			maskpix = np.where(mask!=hp.UNSEEN)[0]
+		else:
+			maskpix = hp.ring2nest(nside, np.where(mask!=hp.UNSEEN)[0])
+
+		#fractional coverage
+		frac = hsp.HealSparseMap.make_empty(32, nside, dtype=np.float64)
```

Comment:
Will this always be 32?  If so maybe make it a constant at the top of the file. If not then an option.

Reply:
added nside_coverage as a config option

Reply:
There's a few other places where 32 is hard-coded - just checking if they should also be set from the config option?

Reply:
yes, sorry i hadn't pushed a few changes, should be there now